### PR TITLE
[IE CLDNN] Add description for  MAX_NUM_THREADS

### PR DESCRIPTION
**Pull request by yeonbok**
### Details:
 - Added description for a new CLDNN plugin configuration : MAX_NUM_THREADS

### Tickets:
 - 44412

### CODE REVIEW

File: docs/IE_DG/supported_plugins/CL_DNN.md [hunk 1]
```diff
@@ -115,7 +115,7 @@ When specifying key values as raw strings (that is, when using Python API), omit
...
 | `KEY_EXCLUSIVE_ASYNC_REQUESTS` | `YES` / `NO`                | `NO`              | Forces async requests (also from different executable networks) to execute serially.|
-
+| `KEY_CLDNN_MAX_NUM_THREADS` | `positive integer` | `maximum # of threads available in host environment` |  Specifies a number of CPU threads that can be used for clDNN engine, e.g, JIT compilation of clDNN kernels or clDNN cpu kernel processing. The default value is set as the number of maximum available threads in host environment to minimize the time for LoadNetwork, where the clDNN kernel build time occupies a large portion. It can be specified with a smaller number according to the usage scenario. For example, if # of availabel threads are 8 and the user wants to utilize 6 threads for other purpose (e.g., running CPU plugin) while clDNN plugin's LoadNetwork is being executed, the value can be set as 2. Note that if the specified value is larger than the maximum available # of threads, it is set as maximum available # of thereads. |
```
[the remark this answers is not in the document]
Comment (isanghao) [reply docs/IE_DG/supported_plugins/CL_DNN.md]: But you mentioned `clDNN cpu kernel processing`? That's why I added that description.

[the remark this answers is not in the document]
Comment (yeonbok) [reply docs/IE_DG/supported_plugins/CL_DNN.md]: Updated!

**Comment by isanghao** [comment]
Looks good to me. 乃

File: docs/IE_DG/supported_plugins/CL_DNN.md [hunk 2]
```diff
@@ -115,7 +115,7 @@ When specifying key values as raw strings (that is, when using Python API), omit
...
 | `KEY_EXCLUSIVE_ASYNC_REQUESTS` | `YES` / `NO`                | `NO`              | Forces async requests (also from different executable networks) to execute serially.|
-
+| `KEY_CLDNN_MAX_NUM_THREADS` | `positive integer` | `maximum # of HW threads available in host environment` |  Specifies the number of CPU threads that can be used for clDNN engine, e.g, JIT compilation of clDNN kernels or clDNN cpu kernel processing. The default value is set as the number of maximum available threads in host environment to minimize the time for LoadNetwork, where the clDNN kernel build time occupies a large portion. Note that if the specified value is larger than the maximum available # of threads, it is set as maximum available # of threads. It can be specified with a smaller number than the available HW threads according to the usage scenario, e.g., when the user wants to assign more CPU threads while clDNN plugin is running. Note that setting this value with lower number will affect not only the network loading time but also the cpu layers of clDNN networks that are optimized with multi-threading. |
```
Comment (vladimir-paramuzov) [inline docs/IE_DG/supported_plugins/CL_DNN.md:118]: You wrote that expected value is positive integer, but the code accepts negative/zero values as well:
```
                int val_i = std::stoi(val);
                if (val_i <= 0 || val_i > max_threads) {
                    n_threads = max_threads;
```

Comment (yeonbok) [reply docs/IE_DG/supported_plugins/CL_DNN.md]: Fixed

**Review by vladimir-paramuzov — APPROVED** [review]

File: docs/IE_DG/supported_plugins/CL_DNN.md [hunk 3]
```diff
@@ -115,7 +115,7 @@ When specifying key values as raw strings (that is, when using Python API), omit
...
 | `KEY_EXCLUSIVE_ASYNC_REQUESTS` | `YES` / `NO`                | `NO`              | Forces async requests (also from different executable networks) to execute serially.|
-
+| `KEY_CLDNN_MAX_NUM_THREADS` | `integer value` | `maximum # of HW threads available in host environment` |  Specifies the number of CPU threads that can be used for clDNN engine, e.g, JIT compilation of clDNN kernels or clDNN cpu kernel processing. The default value is set as the number of maximum available threads in host environment to minimize the time for LoadNetwork, where the clDNN kernel build time occupies a large portion. Note that if the specified value is larger than the maximum available # of threads or less than zero, it is set as maximum available # of threads. It can be specified with a smaller number than the available HW threads according to the usage scenario, e.g., when the user wants to assign more CPU threads while clDNN plugin is running. Note that setting this value with lower number will affect not only the network loading time but also the cpu layers of clDNN networks that are optimized with multi-threading. |
```
[the remark this answers is not in the document]
Comment (yeonbok) [reply docs/IE_DG/supported_plugins/CL_DNN.md:118]: e.g. is a formal representation which is even being used in the academic papers. Why is it expected to be fixed? (for consistency?) BTW this PR is already merged...

**Pull request merged**